File: 42.py
from math import sqrt

words = list(map(lambda x: x[1: -1].lower(), open("p042_words.txt", 'r').read().split(",")))
ALPHABET = list("abcdefghijklmnopqrstuvwxyz")
num_triangles = 0
for word in words:
    k = sum([ALPHABET.index(x) + 1 for x in word])
    test = (-1 + sqrt(1 + 8*k))/2
    if int(test) == test:
        num_triangles += 1

print(num_triangles)

File: cpp/42.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <sstream>
#include <math.h>
#include <cmath>

using namespace std;


vector<string> split(const string s, char delimiter=',')
{
   vector<string> tokens;
   string token;
   istringstream tokenStream(s);
   while (getline(tokenStream, token, delimiter))
   {
      tokens.push_back(token.substr(1, token.length() - 2));
   }
   return tokens;
}


int main() {
    fstream file("42.txt");
    string words;

    getline(file, words);

    vector<string> vect_words = {};
    vect_words = split(words);

    int num_triangles {};

    int s {};
    float test {};
    for (string word: vect_words){
        s = 0;
        for (char c: word)
            s += c - 64;
        
        test = (-1 + sqrt(1 + 8 * s)) / 2;

        if (ceil(test) == test)
            num_triangles++;

    }

    cout << num_triangles << endl;

    return 0;
}